### PR TITLE
Update QuantumEspresso to 7.0

### PR DESCRIPTION
**Pull request by mfherbst**

### CODE REVIEW

**Comment by mfherbst** [comment]
It doesn't seem to like gfortran3 any more ...

File: Q/QuantumEspresso/build_tarballs.jl [hunk 1]
```diff
@@ -80,4 +80,4 @@ dependencies = [
 
 # Build the tarballs, and possibly a `build.jl` as well
 build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies;
-               preferred_gcc_version=v"5", julia_compat="1.6", preferred_llvm_version=v"11")
+               preferred_gcc_version=v"5", julia_compat="1.6")
```
Comment (giordano) [inline Q/QuantumEspresso/build_tarballs.jl:83]: Let's see whether gcc 6 makes any difference:
```suggestion
               preferred_gcc_version=v"6", julia_compat="1.6")
```

Comment (mfherbst) [reply Q/QuantumEspresso/build_tarballs.jl]: yeah, good idea.

**Comment by mfherbst** [comment]
Any clue on that weird real versus complex issue. In particular I'm surprised to only see it for some libgfortran versions on aarch64-apple.

**Comment by giordano** [comment]
For aarch64-apple-darwin [the error](https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=16421&view=logs&j=177558ba-96a5-5472-3281-acd2b316ecb4&t=592f4724-25c7-5af5-ca54-99f8bf35cfe0&l=10015) is
```
[06:50:23] ld: library not found for -lscalapack
[06:50:23] collect2: error: ld returned 1 exit status
```
which makes sense since at the moment we don't have `SCALAPACK_jll` for that platform.  Same thing for armv6l platforms.  I think we have to filter those platforms out until we build `SCALAPACK_jll` there.

**Comment by mfherbst** [comment]
Strange, because it works ok with the other libgfortran versions.

**Comment by giordano** [comment]
Not sure what "other libgfortran versions" you're referring to.

File: Q/QuantumEspresso/build_tarballs.jl [hunk 2]
```diff
@@ -30,8 +30,10 @@ if [ "${nbits}" == 64 ]; then
     flags+=(--with-libxc=yes --with-libxc-prefix=${prefix})
 fi
 
-if [[ "${target}" == powerpc64le-linux-gnu ]]; then
-    # No scalapack binary available on PowerPC
+if [[    "${target}" == powerpc64le-linux-gnu 
+      || "${target}" == armv6l-linux-musleabihf \
+      || "${target}" == armv6l-linux-gnueabihf ]]; then
```
Comment (giordano) [inline Q/QuantumEspresso/build_tarballs.jl:35]:
```suggestion
      || "${target}" == armv6l-linux-* ]]; then
```

**Comment by mfherbst** [comment]
Ah, sorry my bad ... there only is one libgfortran for aarch64-apple-darwin ...

File: Q/QuantumEspresso/build_tarballs.jl [hunk 3]
```diff
@@ -30,8 +30,10 @@ if [ "${nbits}" == 64 ]; then
     flags+=(--with-libxc=yes --with-libxc-prefix=${prefix})
 fi
 
-if [[ "${target}" == powerpc64le-linux-gnu ]]; then
-    # No scalapack binary available on PowerPC
+if [[    "${target}" == powerpc64le-linux-* \
+      || "${target}" == armv6l-linux-* \
+      || "${target}" == aarch64-apple-darwin ]]; then
```
Comment (giordano) [inline Q/QuantumEspresso/build_tarballs.jl:35]:
```suggestion
      || "${bb_full_target}" == armv6l-linux-* \
      || "${target}" == aarch64-apple-darwin* ]]; then
```

**Comment by giordano** [comment]
```
[19:47:58] f951: internal compiler error: in doloop_contained_procedure_code, at fortran/frontend-passes.c:2464
```
I don't think there is much hope for aarch64-apple-darwin at the moment

**Comment by mfherbst** [comment]
I agree ... that might sound stupid, but how do I do that without removing the other apple targets as well?

**Pull request merged**